CONTRIBUTING.md

```diff
@@ -1,3 +1,542 @@
+
+### Beginner Issues
+
+If you are new to the project and want to get into the code, we recommend picking an issue with the label "good first issue". These issues should only require general programming knowledge and little to none insights into the project.
+
+## Requirements
+
+Before you get started you will need a few things installed depending on your operating system.
+
+- OS Package Manager
+- Python 3.6+
+- git
+
+### OSes
+
+We intend to provide first class support for dev setup in the current versions of:
+
+- 🐧 Ubuntu
+- 🍎 MacOS
+- 💠 Windows
+
+If there are missing instructions on setup for a specific operating system or tool please open a PR.
+
+### Linux
+
+If you are using Ubuntu this is `apt-get` and should already be available on your machine.
+
+### MacOS
+
+On macOS, the main package manager is called [Brew](https://brew.sh/).
+
+Install Brew with:
+
+```
+$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+```
+
+Afterwards, you can now use the `brew` package manager for installing additionally required packages below.
+
+### Windows
+
+For Windows users, the recommended package manager is [chocolatey](https://chocolatey.org/).
+Alternatively, you can use [Windows Subsystem Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10) and follow the instructions under Linux section.
+
+## Git
+
+You will need git to clone, commit and push code to GitHub.
+
+### Linux
+
+```
+sudo apt install git
+```
+
+### MacOS
+
+```
+$ brew install git
+```
+
+## Python Versions
+
+This project supports Python 3.6+, however, if you are contributing it can help to be able to switch between python versions to fix issues or bugs that relate to a specific python version. Depending on your operating system there are a number of ways to install different versions of python however one of the easiest is with the `pyenv` tool. Additionally, as we will be frequently be installing and changing python packages for this project we should isolate it from your system python and other projects you have using a virtualenv.
+
+### MacOS
+
+Install the `pyenv` tool with `brew`:
+
+```
+$ brew install pyenv
+```
+
+### For Linux/WSL based Systems
+
+Follow the instructions provided in the repo : https://github.com/pyenv/pyenv-installer
+
+### Using pyenv
+
+Running the command will give you help:
+
+```
+$ pyenv
+```
+
+Lets say you wanted to install python 3.6.9 because its the version that Google Colab uses and you want to debug a Colab issue.
+
+First, search for available python versions:
+
+```
+$  pyenv install --list | grep 3.6
+...
+3.6.7
+3.6.8
+3.6.9
+3.6.10
+3.6.11
+3.6.12
+```
+
+Wow, there are lots of options, lets install 3.6.
+
+```
+$ pyenv install 3.6.9
+```
+
+Now, lets see what versions are installed:
+
+```
+$ pyenv versions
+3.5.9
+3.6.9
+3.7.8
+3.9.0
+```
```

Comment:
though it's just an example can you replace this with python 3.9 ? also we don't even test it for 3.6

Reply:
Done, I also changed some places where it was told that python 3.6 is supported to python 3.7, which is the minimum version that is automatically tested.